nntrainer/tensor/manager.cpp

```diff
@@ -287,24 +339,57 @@ void Manager::untrackLayerInOuts(const std::string layer_name) {
  * @brief Initialize the inputs/outputs for the layer
  */
 void Manager::initializeInOuts(bool trainable) {
+  // Allocate shared derivative memory
   Tensor shared_deriv;
-  if (max_derivative_size > 0 && enable_activation_memory_opt)
+  if (max_derivative_size > 0 && enable_activation_memory_opt && trainable)
     shared_deriv = Tensor(max_derivative_size);
 
+  // @todo Do not count memory of the input tensor of the input layer in the
+  // estimate of max_shared_inout as it is not used
+
+  // Allocate shared input/output memory for inference
+  // @note Memory for label is not allocated here as inference doesnt has label
+  Tensor shared_inout;
+  if (!trainable && enable_inference_inout_memory_opt)
+    shared_inout = Tensor(max_shared_inout);
+
+  // TODO: do not allocate memory for flatten layer
+
   size_t count = 0;
+  bool use_first_last = 0;
```

Comment:
Could you add comment here? I understood from the offline explanation but we might forget later :p

Reply:
done.